### ISSUE TITLE
Projection._project_polygon() assumes that Projection.boundary runs clockwise

### ISSUE THREAD

**Issue by mciura** [SciTools/cartopy#696]
Please take a look at http://stackoverflow.com/questions/33385801/cannot-crop-lakes-with-cartopy

The root of my problem was that the exception is raised from a deep, barely related call, giving no clue to the real cause. I would suggest one of the following:
1) Make Projection.ccw_boundary and Projection.cw_boundary choke on boundaries with wrong orientation and document this behaviour, or
2) Make Projection.ccw_boundary and Projection.cw_boundary detect the orientation of the boundary and always return its properly oriented incarnation.

**Comment by pelson** [comment]
I don't think it is practical to implement a CCW check on the boundary geometry because of the cost of such a check. However, the situation can certainly be improved. A couple of ideas:

 * improve the docs
 * allow either orientation, and a flag on the CRS along the lines of ```_boundary_is_CCW```. For all of the standard projections, this would be set to a constant, but for subclasses of CRS it could be determined at runtime (problems with subclasses of standard CRSes though)

Are you interested in submitting Polyconic as a stock cartopy projection @mciura?

**Comment by mciura** [comment]
Polyconic was just an example, I do not use it anymore.

As for the cost of CCW check, thanks to caching, it would only be paid once. At any rate, it is linear in the number of vertices and I do not expect it to be much higher than the cost of reversing their coords. The call is yours.

**Comment by QuLogic** [comment]
Confusingly, it appears the path required of `GeoAxes.set_boundary` should be _counter_-clockwise, because that's what's expected of matplotlib internally.
